os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -82,15 +82,90 @@ static trwifi_scan_list_s *g_scan_list;
 static int g_scan_num;
 extern struct netdev *ameba_nm_dev_wlan0;
 
+#include "freertos/wrapper.h"
+#include "osdep_service.h"
+#define SCAN_TIMER_DURATION 180000
+
+struct ap_scan_list {
+	unsigned int channel;
+	char ssid[TRWIFI_SSID_LEN + 1];
+	unsigned int ssid_length;
+	int rssi;
+	trwifi_ap_auth_type_e ap_auth_type;
+	trwifi_ap_crypto_type_e ap_crypto_type;
+};
+typedef struct ap_scan_list ap_scan_list_s;
+
+static ap_scan_list_s *saved_scan_list;
+static uint32_t scan_number = 0;
+static _timer scan_timer;
+static _mutex scanlistbusy;	//protects shared variables saved_scan_list,scan_number and scan_timer
+
+static void _scan_timer_handler(void *FunctionContext)
+{
+	ndbg("scan Timer expired : release saved scan list\r\n");
+
+	rtw_mutex_get(&scanlistbusy);
+	ndbg("scan Timer expired : sizeof(ap_scan_list_s) =%d, scan_number=%d \r\n", sizeof(ap_scan_list_s), scan_number);

```

Comment:
I requested eric.kim that wi-fi driver code use vddbg instead of ndbg during reviewing. So could you change ndbg to vddbg?

Reply:
Done.